### PR TITLE
if overlay is omitted leave out dependency on mount unit

### PR DESCRIPTION
**Pull request by teemow**
In my local setup with conair containers I don't use the overlay step. docker, etcd and fleet currently fail because of a hardcoded dependency on the mount unit in their templates.

so i passed through if the overlay is used to the templates and left out `Requires,After` if overlay isn't used.

### CODE REVIEW

File: cli/setup_cmd.go [hunk 1]
```diff
@@ -163,7 +164,7 @@ func setupRun(cmd *cobra.Command, args []string) {
 
 		// !useIPTables is used, because when the iptables step is enabled, we want
 		// --iptables=false for the docker daemon.
-		if err := docker.Setup(fsClient, systemdClient, fetchClient, overlayMountPoint, dockerVersion, privateRegistry, !useIPTables, startDaemons); err != nil {
+		if err := docker.Setup(fsClient, systemdClient, fetchClient, overlayMountPoint, dockerVersion, privateRegistry, !useIPTables, startDaemons, useOverlay); err != nil {
```
Comment (teemow) [inline cli/setup_cmd.go]: It's rather ugly to just add another parameter here. I just was too lazy to cleanup and refactor that mess now.

File: steps/docker/docker.go [hunk 2]
```diff
@@ -6,8 +6,8 @@ import (
 	"strings"
 
 	"github.com/giantswarm/yochu/fetchclient"
-	fsPkg "github.com/giantswarm/yochu/fs"
```
Comment (teemow) [inline steps/docker/docker.go]: This has been changed out of consistency with the fleet/etcd steps.

File: steps/etcd/etcd.go [hunk 3]
```diff
@@ -67,6 +67,23 @@ func Setup(fsc *fs.FsClient, sc *systemd.SystemdClient, fc fetchclient.FetchClie
 	return nil
 }
 
+func createEtcdService(fsc *fs.FsClient, useOverlay bool) error {
```
Comment (teemow) [inline steps/etcd/etcd.go]: Aligned the way how the service template is created with how it is done in the docker step.

File: steps/fleet/fleet.go [hunk 4]
```diff
@@ -71,6 +71,23 @@ func Setup(fsc *fs.FsClient, sc *systemd.SystemdClient, fc fetchclient.FetchClie
 	return nil
 }
 
+func createFleetService(fsc *fs.FsClient, useOverlay bool) error {
```
Comment (teemow) [inline steps/fleet/fleet.go]: Aligned the way how the service template is created with how it is done in the docker step.

**Comment by teemow** [comment]
ping @hectorj2f @denderello

**Comment by denderello** [comment]
I can feel the pain to pass the options down to the templates. For now it's the fastest way to get what you want I admit. Later we could think of generating the unit files rather than using static templates.

Codewise LGTM. If it works in your `conair` setup and tests pass let's go for it.

**Pull request merged**